ldap/servers/slapd/plugin.c

```diff
@@ -2680,6 +2680,24 @@ plugin_free(struct slapdplugin *plugin)
     slapi_ch_free((void **)&plugin);
 }
 
+/*
+ * Check if a plugin is critical or not
+ * Returns:
+ *   1 - critical
+ *   0 - non critical
+ */
+static int
+plugin_critical(struct slapdplugin *plugin)
+{
+    char *non_critical_plugins[] = {"GOST_YESCRYPT", NULL };
+    for (int32_t i = 0; non_critical_plugins[i]; i++) {
```

Comment:
I feel this list could also include the RUST plugins (entryuuid, pbdfk2 256 (libpwdchan-plugin), etc).  Not everyone builds with Rust and I think people have complained about it recently.

Reply:
Thanks @mreynolds389  for your review !!
@progier389 also reported this difficulties with RUST plugin. I agree with entryuuid. With libpwdchan I have a doubt because pbkdf2 is our default storage schema, if it does not start most simple authentication will fail so it looks critical to me.

Reply:
The pkbdf2 in pwdchan is openldap compat pbkdf2 not the same as our inbuilt C one.

There will come a point where we have to draw a line and say "rust is required though", and I think that's a seperate issue. 

But for now I'd say pwdchan can be "optional" if you want. It's just a hard requirement for us at suse with openldap to 389 migrations though. And it's up to RH if you want to support openldap password hash imports when you start to migrate people over.

Reply:
To complete what William is saying. That is the PBKDF2_SHA256 implemented by libpwdstorage-plugin which is mandatory. So I think that we can safely mark pwdchan as "optional" and do not prevent the server to start if the plugin cannot be loaded. (That said if there are some #ifdef specific to SUZE we should use conditional compilation to insure the plugin is mandatory in suze)

Reply:
BTW for the plugin relying on Rust I would rather check the plg_libpath rather than the name